dmpr.py

```diff
@@ -456,224 +432,213 @@ def _recalculate_routing_table(self):
         neigh_routing_paths['othernode_paths']['low_loss'] = dict()
-        for interface_name,interface_data in self._rtd["interfaces"].items():
-            for sender_name,sender_data_raw in interface_data["rx-msg-db"].items():
+        for interface_name, interface_data in self._rtd["interfaces"].items():
+            for sender_name, sender_data_raw in interface_data["rx-msg-db"].items():
                 sender_data = dict()
                 sender_data = sender_data_raw.copy()
                 neigh_routing_paths = self._add_all_neighs(interface_name, interface_data,
                                                            sender_name, sender_data_raw,
                                                            neigh_routing_paths)
-                if len(sender_data['msg']['routingpaths'])>0:
-                   neigh_routing_paths = self._add_all_othernodes_loss(sender_name, sender_data, 
-                                                                       neigh_routing_paths)
-                   neigh_routing_paths = self._add_all_othernodes_bw(sender_name, sender_data, 
-                                                                     neigh_routing_paths)
+                if len(sender_data['msg']['routingpaths']) > 0:
+                    neigh_routing_paths = self._add_all_othernodes_loss(sender_name, sender_data,
+                                                                        neigh_routing_paths)
+                    neigh_routing_paths = self._add_all_othernodes_bw(sender_name, sender_data,
+                                                                      neigh_routing_paths)
         self.log.debug(neigh_routing_paths)
         return neigh_routing_paths
 
-
     def _add_all_neighs(self, interface_name, interface_data, sender_name, sender_data, neigh_routing_paths):
         found_neigh = False
         if len(neigh_routing_paths['neighs']) > 0:
-           for neigh_name, neigh_data in neigh_routing_paths['neighs'].items():
-               if neigh_name == sender_name:
-                  path_found = False
-                  for neigh_path in neigh_data['paths']["{}>{}".
-                                                        format(self._conf["id"], neigh_name)]:
-                      if neigh_path == interface_name:
-                         path_found = True
-                         break
-                  if path_found == False:
-                     neigh_data['paths']["{}>{}".
-                                         format(self._conf["id"],neigh_name)].append(interface_name)
-                  found_neigh = True
-                  break
-           if found_neigh == False:
-              neigh_routing_paths = self._add_neigh_entries(interface_name, sender_name, 
-                                                            sender_data, neigh_routing_paths)
+            for neigh_name, neigh_data in neigh_routing_paths['neighs'].items():
```

Comment:
great improvement, but I think we can improve it even further. I.e. split in separate functions, etc ..

Reply:
see next pr coming soon